src/Generation/generator.py

```diff
@@ -1,62 +1,63 @@
-        examples = json.load(open(EXAMPLES_FILE.format(type_name=type_name), 'r'))
-        self._examples = []
-        for example in examples[EXAMPLES_KEY]:
-            self._examples.append(json.dumps(example, indent=4))
-
-    def generate(self, **kwargs) -> (bool, dict):
-        query_text = self._template_query_text.replace("{{form}}", json.dumps(self._form().model_dump(), ensure_ascii=False, indent=4))
-        for (key, value) in kwargs.items():
+        self._model = getattr(module, _MODEL_CLASS_NAME)
+        with open(_QUERY_FILE.format(type_name=type_name), 'r') as f:
+            self._template_query_text = f.read()
+
+        try:
+            with open(_EXAMPLES_FILE.format(type_name=type_name), 'r') as f:
+                self._examples = list([self._model(**sample) for sample in json.load(f)])
+        except FileNotFoundError:
+            self._examples = []
+
+        try:
+            with open(_PREDEFINED_FILE.format(type_name=self._type_name), 'r') as f:
+                self._fallbacks = list([self._model(**fallback) for fallback in json.load(f)])
+        except FileNotFoundError:
+            self._fallbacks = []
+
+    def generate(self, count: int, **kwargs) -> (bool, dict):
+        query_text = self._template_query_text
+        for key, value in kwargs.items():
             query_text = query_text.replace(f"{{{{{key}}}}}", value)
         
-        if (len(self._examples) > 0):
-            query_text += "\nFor example:\n"
-            for example in self._examples:
-                query_text += "\n" + example
-
-        for _ in range(NUMBER_OF_TRIES_TO_GENERATE):
-            data = self.__generate(query_text)
-            if data != None: return (True, data)
-        
-        # Use predefined
-        return (False, json.load(open(PREDEFINED_FILE.format(type_name=self._type_name))))
-
-    def __generate(self, text: str):
-        generated_text = connector.complete(text, None)[0]
-        json_texts = Generator.__json_pattern.findall(generated_text)
-        # Hopefully it only contains one json substring
-        for json_text in json_texts:
-            json_object = json.loads(json_text)
-            try:
-                form = self._form(**json_object)
-                return form.model_dump()
-            except:
-                pass
-        return None
+        query_parts = query_text.split('---', maxsplit=1)
+
+        try:
+            # Try to get LLM generate us what we want
+            reply = llm.generate(query_parts[0], query_parts[1] if len(query_parts) > 0 else 'Answer:', self._model, self._examples, count)
+            return (True, reply)
+        except llm.GenerateFailure as e:
+            if len(self._fallbacks) > 0:
+                # Use predefined fallback
+                return (False, self._fallbacks[random.randint(0, len(self._fallbacks) - 1)].model_dump())
+            else:
+                # No predefined options available, need to crash
+                raise e
```

Comment:
Why is not available? :(

Reply:
This would mean missing predefined.json file, I'll clarify this in comment.

Reply:
Yes, yes, I understand that. Sorry, I put the question in the wrong place. It was more about whether it wouldn't be better to generate a predefined answer for everything with, for example chatgpt, just to be safe. Mostly because of cohere, I dont trust that one :D

Reply:
I'll keep in mind as future development, but the current system does not make it easy to run multiple LLMs at same time :/